src/backend/distributed/metadata/node_metadata.c

```diff
@@ -1631,8 +1636,12 @@ SetWorkerColumn(WorkerNode *workerNode, int columnIndex, Datum value)
 
 	table_close(pgDistNode, NoLock);
 
-	/* we also update the column at worker nodes */
-	SendCommandToWorkersWithMetadata(metadataSyncCommand);
+	if (!localOnly && columnIndex != Anum_pg_dist_node_hasmetadata)
```

Comment:
As I noted, we should not have this trick. Is this related to calling the function in `SetUpDistributedTableDependencies()`? Lets first remove that and see the situation.